### ISSUE TITLE
An object reference is required in nameof in attribute usage

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#40229]
**Version Used**: master

**Steps to Reproduce**: [sharplab](https://sharplab.io/#v2:D4AQzABCBMEPIBcAWBTAThAsAKAN44kKkgGEAbAQwGcqJzqqBuHAXx3Clnpp32yIgBtAIIIEaABQA7CgFsUAewBmExKjQA6blQCUOgLoQA9EYjo0CjCQDKABgCM0WxGFSICgEYArFAGMEEGgoSugoUr4oEACWtEEAjgCuUUEAJgREHGro8MjozNhs2BwwLmIYAFxQ9hqi4lEeCQgovOmEHLWSIPbOuhC4hSxAA==)

```cs
public class Other 
{
    public Class Class;
}
public class Class
{
    [Attr(nameof(Other.Class))] // error CS0120 An object reference is required
    public Other Other;
}
public class Attr : System.Attribute
{
    public Attr(string s) {}
}
```

**Expected Behavior**: no error

**Actual Behavior**: `error CS0120 An object reference is required`

**Comment by alrz** [comment]
this could be a color color situation but whether Other binds to the member or type, nameof should not produce an error.

**Comment by svick** [comment]
This doesn't seems to be caused by the Color Color problem, [the following code still doesn't compile](https://sharplab.io/#v2:D4AQzABCBMEPIFgBQBvZENUgYQtgNgIYDOxA3MgL7LhSzbJpKYQDaAggC6cBOAFADtCAWwCmAewBmfOJwAWongDoCJYgEp1AXQgB6XREU9xPPAGUADAEZoFiOwERxAIwBWogMacIPUZMWiAh6iEACWxD6iAI4ArqG+ACbomLRw8PKKFEjUSLQw9tymAFxQVkpcvKHOMZyijMkYtBX8IFZ2GhAoOZRAA=):

```c#
public class O
{
    public C Class;
}
public class C
{
    [Attr(nameof(Other.Class))] // error CS0120 An object reference is required
    public O Other;
}
```

**Comment by jcouv** [comment]
In @svick's example, the error seems expected. It should use `nameof(O.Class)`.
In @alrz's example then, the error seems expected too. When looking up what `Other` means, we'll find the field named `Other` first (but accessing it requires an instance). See [lookup rules](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#simple-names).

**Comment by alrz** [comment]
>In @alrz's example then, the error seems expected too. When looking up what Other means, we'll find the field named Other first (but accessing it requires an instance). 


That's true but we should not get an error here [as mentioned in the spec](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#nameof-expressions), 

> However, where the lookup described in Simple names and Member access results in an error because an instance member was found in a static context, a *nameof_expression* produces no such

**Comment by jcouv** [comment]
Ah, thanks for the pointer. Bug indeed!  ;)

**Comment by YairHalberstadt** [comment]
Slightly simpler reproduction:

```csharp
public class C {
        public string S { get; private set; }
        public static string M() => nameof(S.Count);
}
```

**Comment by BillWagner** [comment]
@jcouv From the discussion, it looks like the resolution for this is to change the compiler, not the spec. If that's correct, I won't move it to the dotnet/csharpstandard repo.
